### ISSUE TITLE
Better way of enabling colors for processes launched from IDE

### ISSUE THREAD

**Issue by segrey** [sindresorhus/supports-color#13]
Hi,

A Node.js process launched from IDE has `process.stdout.isTTY` always set to `false`, because the Node.js process is started with redirected streams. Unfortunately, AFAIK, there is no way to set it to `true`.
So, to enable colors users should specify `--color` command line flag manually.
IDE can improve the situation and add `--color` flag automatically. But it seems a bit unsafe for me, because it isn't clear where the safe place for `--color` flag is (specifying `--color` as the first argument might be incorrect for some apps and specifying it as the last one might be incorrect for others).

Possibly safer way would be to introduce a new environment variable (like `ENABLE_ANSI_COLORS`) and check for it before checking for `process.stdout.isTTY`:
```javascript
	if (/^1|^true|^yes|always/i.test(process.env.ENABLE_ANSI_COLORS)) {
		return true;
	}
	if (process.stdout && !process.stdout.isTTY) {
		return false;
	}
```

Another way is to place check for `COLORTERM` environment variable before checking for `process.stdout.isTTY`, but I'm not sure it wouldn't break some use-cases.

Thanks.

**Comment by sindresorhus** [comment]
Maybe `SUPPORTS_COLOR in process.env` then. Still not convinced this is a good idea though.

**Comment by segrey** [comment]
Maybe it isn't a good idea. I just don't see any other ways to allow IDE to enable color support at the moment. Also, introducing `SUPPORTS_COLOR` shouldn't harm. It can be marked as experimental to make it easier to remove it in future if some better solution would be figured out.
Another name: `SUPPORTS_COLOR_NO_TTY` to indicate that current process doesn't have TTY device associated with it, but it's capable to color output.

**Comment by bguiz** [comment]
My 2 cents - I have a use case where environment variables are possible, but command line flags are not. PR seems straightforward enough: #17

**Comment by sindresorhus** [comment]
What kind of use-case?

**Comment by bguiz** [comment]
The use case is that I am using `child_process.exec` to run a another module in a separate process. This module rejects the `--color` CLI parameter, as it is strict about which flags it does and does not allow, and it has good reasons for doing so. 

This module needs to invoke chalk, which in turns invoked supports-color. I have no reasonable means to communicate a flag other than through an environment variable, and hence making supports-color support this, I believe, is the right place for the fix to go.

**Comment by bguiz** [comment]
My current workaround is

```javascript
var chalk = require('chalk');
if (!!process.env.SUPPORTS_COLOR) {
  chalk.enabled = function() { return true; };
}
```

Which is, for obvious reasons, not ideal.

**Comment by sindresorhus** [comment]
@bguiz Yeah, I see the need. Just keep in mind that whatever I go with here I'll be stuck with forever, so I want to get it right.

I'm not sure `SUPPORTS_COLOR` is the right name either.

Some alternatives: `COLOR`, `FORCE_COLOR`.

**Comment by bguiz** [comment]
Of course:

```javascript
var choices = ['SUPPORTS_COLOR', 'COLOR', 'FORCE_COLOR', 'FORCE_SUPPORTS_COLOR'];
```

I am a fan of having `'SUPPORTS_COLOR'` somewhere in the environment variable, because that minimises the chances of overlap with an environment variable used by something else.

**Comment by segrey** [comment]
`SUPPORTS_COLOR` is fine. The only con that occurs to me is that the name is still a bit general, so there is a small probability of overlapping with other environment variable. An example of more unique name could be `(SUPPORS|ENABLE)_ANSI_COLOR_FOR_(NO_TTY|REDIRECTED_STREAMS)` (addresses only this use-case and nothing more).
Seems the problem with `FORCE_*COLOR*` names is that it isn't clear what would happen if a CLI flag `--no-color` and `FORCE_*COLOR*` are both specified.

**Comment by jbnicolai** [comment]
I'm leaning towards `FORCE_COLOR`, since we're overwriting all other default behaviour. No disabling when running in a non-tty/pipe. No respecting the CLI flags. Etc..

**Comment by segrey** [comment]
>No disabling when running in a non-tty/pipe. No respecting the CLI flags.

Yes, that would be fair.

**Comment by rmg** [comment]
It can be very useful to output ANSI escape sequences like colours to a pipe, like when redirecting to a pager that supports raw mode. 

TTY checks are generally more appropriate for inputs than outputs.

If you were doing more elaborate output like curses/ncurses, you'd want to look at TERM to determine which terminal emulator to target.

**Comment by simcha** [comment]
I would also vote +1. My use case is Ember CLI + Atom.io editor. Ember is passing all options to registered extensions so changing Ember would brake contract with extensions. I do not know if they are willing to do so.

**Comment by sindresorhus** [comment]
fixed by #18
